app/src/androidTest/java/com/github/campus_capture/bootcamp/FirebaseBackendTest.java

```diff
@@ -0,0 +1,262 @@
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+import android.util.Log;
+
+import androidx.annotation.NonNull;
+import androidx.test.core.app.ApplicationProvider;
+import androidx.test.espresso.action.ViewActions;
+import androidx.test.espresso.intent.Intents;
+import androidx.test.espresso.matcher.ViewMatchers;
+import androidx.test.ext.junit.rules.ActivityScenarioRule;
+import androidx.test.ext.junit.runners.AndroidJUnit4;
+
+import com.github.campus_capture.bootcamp.AppContext;
+import com.github.campus_capture.bootcamp.R;
+import com.github.campus_capture.bootcamp.authentication.Section;
+import com.github.campus_capture.bootcamp.firebase.BackendInterface;
+import com.github.campus_capture.bootcamp.firebase.FirebaseBackend;
+import com.github.campus_capture.bootcamp.firebase.PlaceholderFirebaseInterface;
+import com.github.campus_capture.bootcamp.scoreboard.ScoreItem;
+import com.google.android.gms.tasks.OnCompleteListener;
+import com.google.android.gms.tasks.Task;
+import com.google.firebase.database.DataSnapshot;
+import com.google.firebase.database.DatabaseError;
+import com.google.firebase.database.DatabaseReference;
+import com.google.firebase.database.FirebaseDatabase;
+import com.google.firebase.database.ServerValue;
+import com.google.firebase.database.ValueEventListener;
+
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Ignore;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.security.spec.ECField;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+
+@RunWith(AndroidJUnit4.class)
+public class FirebaseBackendTest {
+
+    static FirebaseDatabase database = null;
+
+    @BeforeClass
+    public static void init_firebase_emulator() {
+
+        try{
+            AppContext context = AppContext.getAppContext();
+            database = context.getFirebaseDB();
+            Log.d("FirebaseBackendTest", "enable emulator");
+            database.useEmulator("10.0.2.2", 9000);
+
+        } catch (Exception e) {
+            Log.e("FirebaseBackendTest", e.toString());
+        }
+    }
+
+    @Before
+    public void clear_firebase_database(){ database.getReference().setValue(null);}
+
+    @Test
+    public void testVoteZone() {
+        // set database content
+        database.getReference().child("Users").child("testUserId").child("has_voted").setValue(false);
+        database.getReference().child("Zones").child("BC").child("IN").setValue(4);
+
+        BackendInterface b = new FirebaseBackend();
+
+        try {
+            Boolean result = b.voteZone("testUserId", Section.IN, "BC").get();
+            assertTrue(result);
+        }catch(Exception e){
+            Log.e("Error in test", e.toString());
+            assertTrue(false);
```

Comment:
As I've said, you can replace this with `fail()` :)

Reply:
ah yes I forgot one